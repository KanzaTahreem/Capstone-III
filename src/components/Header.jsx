import { useDispatch } from 'react-redux';
import { BiSearch } from 'react-icons/bi';
import { userInput } from '../redux/AllCoins/coins';
import headerImg from '../assets/images/header.png';
import '../styles/Header.css';

const Header = () => {
  const dispatch = useDispatch();

  const handleSearch = (e) => {
    dispatch(userInput(e.target.value));
  };

  return (
    <header>
      <div className="inner_header">
        <div>
          <h2>Hi, there ğŸ‘‹</h2>
          <p>Ready to start your day with some coins?</p>
          <form>
            <input type="text" name="name" id="name" placeholder="Search here..." onChange={handleSearch} data-testid="input" />
            <BiSearch />
          </form>
        </div>
        <img src={headerImg} alt="girl working on laptop" />
      </div>
    </header>
  );
};

export default Header;
