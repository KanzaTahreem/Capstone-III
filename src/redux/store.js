import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './AllCoins/coins';
import chartReducer from './CoinChart/chart';

const store = configureStore({
  reducer: {
    crypto: coinsReducer,
    chart: chartReducer,
  },
});

export default store;
