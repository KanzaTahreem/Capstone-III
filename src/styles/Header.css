.header-container {
  gap: 5em;
  height: 100vh;
  width: 100%;
  display: flex;
  padding: 15em 0;
  text-align: center;
  flex-direction: column;
  background-size: cover;
  background-position: 50%;
  background-attachment: fixed;
  background-repeat: no-repeat;
  box-shadow: inset 0 -9px 35px rgb(0 0 0 / 25%);
  background-image: linear-gradient(rgba(0, 0, 33, 0.8), #3033b9), url('../assets/images/bg.jpg');
}

.quote {
  color: #eee;
  font-size: 3em;
  font-weight: bold;
  text-shadow: 2px -5px 0 rgb(0 0 0 / 55%);
  filter: drop-shadow(1px 2px 4px black);
}

#name {
  box-shadow: 0 8px 6px -5px black;
  width: 40%;
  color: #fff;
  height: 7vh;
  border: 1px solid #aaa;
  background: transparent;
  padding: 1em;
  border-radius: 2em;
  filter: drop-shadow(2px 4px 6px black);
}

#name:focus {
  outline: none;
  color: #fff;
  letter-spacing: 2px;
  filter: none;
  background-color: rgba(0, 0, 0, 0.122);
}
