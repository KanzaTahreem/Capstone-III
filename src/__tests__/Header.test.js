import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Header from '../components/Header';

it('renders correctly', () => {
  const rocket = renderer
    .create(<Provider store={store}><Header /></Provider>)
    .toJSON();
  expect(rocket).toMatchSnapshot();
});
