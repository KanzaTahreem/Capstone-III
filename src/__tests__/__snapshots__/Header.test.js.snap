// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header-container"
>
  <h2
    className="quote"
    data-testid="quote"
  >
    The future of Money is digital currency
  </h2>
  <form>
    <input
      data-testid="input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Search here..."
      type="text"
    />
  </form>
</div>
`;
